rero_ils/modules/patron_transactions/api.py

```diff
@@ -54,3 +60,153 @@ class PatronTransaction(IlsRecord):
+        from ..notifications.api import Notification
+        if self.notification_pid:
+            notif = Notification.get_record_by_pid(self.notification_pid)
+            return notif.loan_pid
+        return None
+
+    @property
+    def patron_pid(self):
+        """Return the patron pid of the patron transaction."""
+        return self.replace_refs()['patron']['pid']
+
+    @property
+    def notification_pid(self):
+        """Return the notification pid of the patron transaction."""
+        if self.get('notification'):
+            return self.replace_refs()['notification']['pid']
+        return None
+
+    @property
+    def notification(self):
+        """Return the notification of the patron transaction."""
+        from ..notifications.api import Notification
+        if self.get('notification'):
+            pid = self.replace_refs()['notification']['pid']
+            return Notification.get_record_by_pid(pid)
+        return None
+
+    @property
+    def notification_transaction_location_pid(self):
+        """Return the transaction location of the notification."""
+        notif = self.notification
+        if notif:
+            return notif.transaction_location_pid
+        return None
+
+    @property
+    def notification_transaction_user_pid(self):
+        """Return the transaction user pid of the notification."""
+        notif = self.notification
+        if notif:
+            return notif.transaction_user_pid
+        return None
+
+    @classmethod
+    def create_patron_transaction_from_notification(
+            cls, notification, dbcommit, reindex, delete_pid):
+        """Create a patron transaction from notification."""
+        record = {}
+        if notification.get('notification_type') == 'overdue':
+            data = build_patron_transaction_ref(notification, {})
+            data['creation_date'] = datetime.now(timezone.utc).isoformat()
+            data['type'] = 'overdue'
+            data['status'] = 'open'
+            record = cls.create(
+                data,
+                dbcommit=dbcommit,
+                reindex=reindex,
+                delete_pid=delete_pid
+            )
+        return record
+
+    @property
+    def currency(self):
+        """Return patron transaction currency."""
+        organisation_pid = self.organisation_pid
+        return Organisation.get_record_by_pid(organisation_pid).get(
+            'default_currency')
+
+    @property
+    def events(self):
+        """Shortcut for events of the patron transaction."""
+        events = []
+        results = PatronTransactionEventsSearch().filter(
+            'term', parent__pid=self.pid
+        ).source().scan()
+        try:
+            pid = next(results).pid
+            patron_event = PatronTransactionEvent.get_record_by_pid(pid)
+            events.append(patron_event)
+            return events
+        except StopIteration:
+            return events
```

Comment:
Why not using a simple loop structure with `yield` ?

Reply:
fixed